src/Groups/libraries/perfectgroups.jl

```diff
@@ -178,7 +178,78 @@ function number_of_perfect_groups(n::IntegerUnion)
    return res::Int
 end
 
-# TODO: add all_perfect_groups() iterator
+
+"""
+    all_perfect_groups(L...)
+
+Return the list of all perfect groups (up to permutation isomorphism)
+satisfying the conditions described by the arguments. These conditions
+may be of one of the following forms:
+
+- `func => intval` selects groups for which the function `func` returns `intval`
+- `func => list` selects groups for which the function `func` returns any element inside `list`
+- `func` selects groups for which the function `func` returns `true`
+- `!func` selects groups for which the function `func` returns `false`
+
+As a special case, the first argument may also be one of the following:
+- `intval` selects groups whose order equals `intval`; this is equivalent to `order => intval`
+- `intlist` selects groups whose order is in `intlist`; this is equivalent to `order => intlist`
+
+The following functions are currently supported as values for `func`:
```

Comment:
Note that I do not actually restrict which functions are used in the code; my reading of "supported" here is "we promise these will work, anything else can lead to errors. I could explicitly reject other functions, but there seems no point in this right now.

Reply:
No problem with the code; if someone asks for supersolvable perfect groups then this is fine.
But it is irritating to read `is_supersolvable` here.

Reply:
This was exactly what I expected from the wording.
Is there a reason why this list is so much shorter than e.g. for the transitive groups?
https://github.com/oscar-system/Oscar.jl/blob/83f05750497b654dfabafc210919f88dfbea534c/src/Groups/libraries/transitivegroups.jl#L162-L179

I suppose this is too computationally expensive?

Reply:
No, just most of these filters make no sense: the only perfect group which is abelian/cyclic/nilpotent/solvable/super-solvable is the trivial group.

And primitive/transitive/degree etc. don't really make sense here -- while we return permutation groups, (a) in a future version we may give the user a choice to also get fp groups, (b) which permutation group representations we use is arbitrary. I.e.: the "degree" of a transitive group is well-defined; the "degree" of a perfect group is not.

Reply:
Ah okay, thanks for clarifying this.